### ISSUE TITLE
DataTable ToolbarSearch: long input texts overlap with the 'x' btn.

### ISSUE THREAD

**Issue by EricPyZhou** [carbon-design-system/carbon#5142]
in DataTable with Toolbar, long input texts overlap with the 'x' btn. By looking at its css styles, it could possibly be the padding-right of the input element not being set properly. By comparing with the default search component, I found out that the padding-left both exist, but the padding right only exist in the default search

## Environment

> Operating system

> Browser

> Automated testing tool and ruleset

> Assistive technology used to verify

## Detailed description

> What version of the Carbon Design System are you using?

> What did you expect to happen?

> What happened instead?
![image](https://user-images.githubusercontent.com/26387900/72930987-139a7880-3d2b-11ea-8a0c-c5a2c5c4017f.png)
> What WCAG 2.1 checkpoint does the issue violate?

## Steps to reproduce the issue

1. Step one
2. Step two
3. Step three
4. etc.

> Please create a reduced test case in CodeSandbox
>
> - Style and vanilla JS:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/components/examples/codesandbox
> - React:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/react/examples/codesandbox

## Additional information

- Screenshots or code
- Notes

**Comment by EricPyZhou** [comment]
Thank you! @tw15egan
